src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -57,183 +58,162 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     })
   }
 
+  private def isReservedAttribute(instanceType: InstanceType, attribute: String) = {
+    val alwaysReservedAttributes: Set[String] = Set("space", "externalId", "_type")
+
+    // type is supported as an alias to _type for edges for legacy reasons.
+    val edgeReservedAttributes: Set[String] = Set("type", "startNode", "endNode")
+    alwaysReservedAttributes.contains(attribute) ||
+    (instanceType == InstanceType.Edge && edgeReservedAttributes.contains(attribute))
+  }
+
   private def extractInstancePropertyValue(key: String, value: InstancePropertyValue): Any =
     FlexibleDataModelRelationUtils.extractInstancePropertyValue(schema.apply(key).dataType, value)
 
-  protected def toInstanceFilter(
+  private def getVersionedViewExternalId(viewReference: ViewReference): String =
+    s"${viewReference.externalId}/${viewReference.version}"
+
+  private def getViewAttributeReference(
+      viewReference: ViewReference,
+      propertyName: String): Seq[String] =
+    Seq(viewReference.space, getVersionedViewExternalId(viewReference), propertyName)
+
+  protected def toFilter(
       instanceType: InstanceType,
       sparkFilter: Filter,
-      space: String,
-      versionedExternalId: String): Either[CdfSparkException, FilterDefinition] = {
+      // viewReference is required for non-reserved attribute filters resolution
+      viewReference: Option[ViewReference]
+  ): Either[CdfSparkException, FilterDefinition] =
+    sparkFilter match {
+      case EqualTo(attribute, _) if isReservedAttribute(instanceType, attribute) =>
+        toNodeOrEdgeAttributeFilter(instanceType, sparkFilter)
+      case IsNull(attribute) if isReservedAttribute(instanceType, attribute) =>
+        toNodeOrEdgeAttributeFilter(instanceType, sparkFilter)
+      case IsNotNull(attribute) if isReservedAttribute(instanceType, attribute) =>
+        toNodeOrEdgeAttributeFilter(instanceType, sparkFilter)
+      case StringStartsWith(attribute, _) if isReservedAttribute(instanceType, attribute) =>
```

Comment:
so for /sync case we will now allow pushdown of prefix filters which isn't necessarily backed by indexes, so either we take risk here or make sure we don't pushdown it on sync (on non-sync pushdown was present, so might be a sign index exists, or might be a bug or conscious decision, no pre-existing comments on it...)

Reply:
The most logical thing would be that the index is used, but yes it is a bit of a risk. I would go for it personally.